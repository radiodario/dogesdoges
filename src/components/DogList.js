import React from 'react';
import PropTypes from 'prop-types';
import Dog from './Dog';

const DogList = ({ dogs }) => (
  <ul>
    { dogs.map((dog, index) => (
      <Dog key={index} {...dog} />
    ))}
  </ul>
)

DogList.propTypes = {
  dogs: PropTypes.arrayOf(
    PropTypes.shape({
      breed: PropTypes.string.isRequired,
      subBreed: PropTypes.string,
      text: PropTypes.string.isRequired,
      imageUrl: PropTypes.string.isRequired,
      }).isRequired
    ).isRequired
}

export default DogList

