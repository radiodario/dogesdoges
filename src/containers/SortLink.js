import { connect } from 'react-redux'
import { setSortingDirection } from '../actions'
import Link from '../components/Link'

const mapStateToProps = (state, ownProps) => ({
  active: ownProps.direction === state.sorting
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  onClick: () => dispatch(setSortingDirection(ownProps.direction))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Link)
