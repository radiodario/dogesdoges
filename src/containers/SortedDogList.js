import { connect } from 'react-redux';
import { sortBy } from 'lodash';
import DogList from '../components/DogList';

const getSortedDogs = (dogs, sorting) => {
  switch (sorting) {
    case 'ASC':
      return sortBy(dogs, d => (d.breed + d.subBreed))
    case 'DSC':
      return sortBy(dogs, d => (d.breed + d.subBreed)).reverse()
    default:
      return dogs
  }
}

const mapStateToProps = state => {
  return {
    dogs: getSortedDogs(state.dogs, state.sorting)
  }
}

const SortedDogList = connect(
  mapStateToProps,
)(DogList)

export default SortedDogList
