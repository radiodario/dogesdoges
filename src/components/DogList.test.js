import React from 'react';
import DogList from './DogList';
import renderer from 'react-test-renderer';
import { getSortedDogs } from '../containers/SortedDogList';
import { dogs } from '../containers/StoredDogList.test';

it('renders correctly', () => {
  const tree = renderer
    .create(<DogList dogs={getSortedDogs(dogs)} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

