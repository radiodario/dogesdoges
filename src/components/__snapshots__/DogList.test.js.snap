// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <div
    className="Dog-Breed"
  >
    <h4>
      terrier - boston
    </h4>
    <img
      alt="terrier - boston"
      src="hi"
    />
    <div
      className="Dog-Breed--Text"
    >
      hi
    </div>
  </div>
  <div
    className="Dog-Breed"
  >
    <h4>
      corgi
    </h4>
    <img
      alt="corgi"
      src="hi"
    />
    <div
      className="Dog-Breed--Text"
    >
      hi
    </div>
  </div>
  <div
    className="Dog-Breed"
  >
    <h4>
      mastiff - spanish
    </h4>
    <img
      alt="mastiff - spanish"
      src="hi"
    />
    <div
      className="Dog-Breed--Text"
    >
      hi
    </div>
  </div>
  <div
    className="Dog-Breed"
  >
    <h4>
      mastiff - english
    </h4>
    <img
      alt="mastiff - english"
      src="hi"
    />
    <div
      className="Dog-Breed--Text"
    >
      hi
    </div>
  </div>
</ul>
`;
